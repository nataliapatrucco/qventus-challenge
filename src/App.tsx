import React from "react";
import "./App.css";
import { generateValidationOptions } from "./utils/generateValidations";
import { Validation } from "./components/Validation";

const App: React.FC = () => {
  const selectedValidations = [
    "hasNumbers",
    "hasSpecialCharacters",
    "hasUppercaseLetter",
    "hasNoConsecutiveCharacters",
  ];
  const validations = generateValidationOptions(selectedValidations);

  return (
    <div className="App">
      <h1>Password Component</h1>
      <Validation options={validations} />
    </div>
  );
};

export default App;
