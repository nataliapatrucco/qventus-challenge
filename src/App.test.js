import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

const options = jest.mock("./utils/generateValidations", () => ({
  generateValidationOptions: jest.fn(() => [
    { message: "Test validation 1", validation: () => true },
    { message: "Test validation 2", validation: () => false },
  ]),
}));

describe("App component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
