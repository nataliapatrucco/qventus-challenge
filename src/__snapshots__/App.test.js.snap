// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Password Component
    </h1>
    <div
      style="display: flex; justify-content: center; align-items: center;"
    >
      <input
        aria-label="Password"
        type="password"
      />
      <ul
        style="list-style: none; padding: 0px;"
      >
        <li
          style="margin-bottom: 1rem;"
        >
          <span
            style="background-color: red; color: white; padding: 2rem; clip-path: circle(20%);"
          >
            x
          </span>
          Has a number 0-9
        </li>
        <li
          style="margin-bottom: 1rem;"
        >
          <span
            style="background-color: red; color: white; padding: 2rem; clip-path: circle(20%);"
          >
            x
          </span>
          Has a special char !@#$%^&*
        </li>
        <li
          style="margin-bottom: 1rem;"
        >
          <span
            style="background-color: red; color: white; padding: 2rem; clip-path: circle(20%);"
          >
            x
          </span>
          Has uppercase Letter
        </li>
        <li
          style="margin-bottom: 1rem;"
        >
          <span
            style="background-color: red; color: white; padding: 2rem; clip-path: circle(20%);"
          >
            x
          </span>
          Has no consecutive characters
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
